sync/_databricks.py

```diff
@@ -582,6 +662,61 @@ def get_prediction_cluster(
     return prediction_response
 
 
+def apply_project_recommendation(job_id: str, project_id: str, recommendation_id: str = None):
+    """Updates jobs with prediction configuration
+
+    :param job_id: ID of job to apply prediction to
+    :type job_id: str
+    :param project_id: Sync project ID
+    :type project_id: str
+    :param recommendation_id: Sync prediction ID, defaults to latest in project
+    :type recommendation_id: str, optional
+    :return: ID of applied recommendation
+    :rtype: Response[str]
+    """
+    databricks_client = get_default_client()
+
+    job = databricks_client.get_job(job_id)
+    job_clusters = _get_project_job_clusters(job)
+
+    project_cluster = job_clusters.get(project_id)
+    if not project_cluster:
+        if len(job_clusters) == 1 and None in job_clusters:
+            project_cluster = job_clusters[None]
+        else:
+            return Response(
+                error=DatabricksError(
+                    message=f"Unable to locate cluster in job {job_id} for project {project_id}"
```

Comment:
Should we have a different error if job_clusters is empty? It seems like we return an empty signature if the project has more than one job cluster associated with it

Reply:
> seems like we return an empty signature if the project has more than one job cluster associated with it

That's correct.

Yeah, I think that'd be good to be more explicit about each error scenario. I think this is good enough for now though - at least we have logging in `_get_project_job_clusters` for the scenario when there's more than 1 cluster for a project.